### PR TITLE
[Merged by Bors] - Add Box::from_corners method

### PR DESCRIPTION
**Pull request by komadori**
# Objective

This add a ctor to `Box` to aid the creation of non-centred boxes. The PR adopts @rezural's work on PR #3322, taking into account the feedback on that PR from @james7132.

## Solution

`Box::from_corners()` creates a `Box` from two opposing corners and automatically determines the min and max extents to ensure that the `Box` is well-formed.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_render/src/mesh/shape/mod.rs [hunk 1]
```diff
@@ -49,6 +49,20 @@ impl Box {
...
+    pub fn from_corners(a: Vec3, b: Vec3) -> Box {
+        let max = a.max(b);
+        let min = a.min(b);
+        Box {
+            max_x: max.x,
+            min_x: min.x,
+            max_y: max.y,
+            min_y: min.y,
+            max_z: max.z,
+            min_z: min.z,
```
Comment (james7132) [inline crates/bevy_render/src/mesh/shape/mod.rs:63]: nit: all max, then all min, this is a bit hard to follow IMO.

**Review by james7132 — APPROVED** [review]
Only one nit, otherwise LGTM.

**Review by mockersf — APPROVED** [review]

**Comment by mockersf** [comment]
moving to 0.10 as 0.9.1 as this is a new feature
